File: src/main/java/Student.java
/**
 * @author zengwang
 * @create 2021-10-26 10:39 下午
 * @function:
 */
public class Student {
    private String name;
    private Integer age;
    // 静态变量，全局只有一份，不属于对象，只能用类去调用
    private static String school = "atguigu";

    public Student(String name, Integer age) {
        this.name = name;
        this.age = age;
    }

    public void printInfo(){
        System.out.println(this.name + " " + this.age + " " + Student.school);
    }
    /* 面向对象中，任何方法、属性的调用都应该通过对象调用，而上面的school属性则体现了java不够面向对象
    所以，scala中去掉了static关键字
     */
    public static void main(String[] args) {
        Student alice = new Student("alice", 20);
        Student bob = new Student("bob", 23);
        alice.printInfo();
        bob.printInfo();
    }
}

File: src/main/scala/com/atguigu/chapter01/Student.scala
package com.atguigu.chapter01

/**
 *
 * @author zengwang
 *
 * @create 2021-09-30 4:32 下午
 *
 */
class Student(name: String, age: Int) {
  def printInfo(): Unit = {
    println(name + " " + age + " " + Student.school)
  }
}

// 引入伴生对象
object Student {
  val school: String = "atguigu"

  def main(args: Array[String]): Unit = {
    // 使用的是class Student做的实例化
    val alice = new Student("alice", 20)
    alice.printInfo()
    var name = "ming"
    var age = 20
    println(s"${name} age is ${age}")

    var sql: String = s"""
       |select *
       |from
       |  student
       |where
       |  name = ${name}
       |and
       |  age > ${age}
       |""".stripMargin
    println(sql)
  }
}